src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -1100,14 +1100,21 @@ export const selectStakingOpportunitiesDataFullByFilter = createCachedSelector(
   selectStakingDataByFilter,
   selectAssetIdParamFromFilter,
   selectDefaultStakingDataByValidatorId,
+  selectFeatureFlags,
   (
     portfolioValidatorIds,
     validatorsData,
     stakingDataByValidator,
     assetId,
     defaultStakingData,
+    featureFlags,
   ): OpportunitiesDataFull[] => {
-    if (defaultStakingData) {
+    // used to enable/disable staking by feature flag
+    const featureFlagFilter = () => {
+      if (assetId === osmosisAssetId && !featureFlags.OsmosisStaking) return false
+      return true
+    }
```

Comment:
No need for this to be a function:

```suggestion
    const featureFlagFilter = !(assetId === osmosisAssetId && !featureFlags.OsmosisStaking)
```

The below consumption by the filter looks incorrect as the function takes no args, as if `featureFlagFilter()` evaluates to `false` we are always returning an empty array (so no need to filter if that's the intended outcome).

Reply:
I didn't want to assume we would always return an empty array there, but if that's the case we can remove this.
I didn't really have context on the function of the "dummy" data, so wanted to make sure they got filtered out.  
e3c94a02846299e20e508512ac56a29c7e28863a

Reply:
It would always be false so my thinking was incorrect